crates/net/rpc-types/src/admin.rs

```diff
@@ -0,0 +1,72 @@
+use reth_primitives::{NodeRecord, U256};
+use serde::{Deserialize, Serialize};
+
+/// Represents the `admin_nodeInfo` response, which can be queried for all the information
+/// known about the running node at the networking granularity.
+///
+/// Note: this format is not standardized.  Reth follows geth's format,
+/// see: https://geth.ethereum.org/docs/interacting-with-geth/rpc/ns-admin
+#[derive(Serialize, Deserialize, Debug)]
+pub struct NodeInfo {
+    /// Enode in URL format.
+    pub enode: String,
+    /// ID of the local node.
+    pub id: String,
+    /// IP of the local node.
+    pub ip: String,
+    /// Address exposed for listening for the local node.
+    #[serde(rename = "listenAddr")]
+    pub listen_addr: String,
+    /// Local node client name.
+    pub name: String,
+    /// Ports exposed by the node for discovery and listening.
+    pub ports: Ports,
+    /// Networking protocols being run by the local node.
+    pub protocols: Protocols,
+}
+
+impl NodeInfo {
+    /// Creates a new instance of `NodeInfo`.
+    pub fn new(enr: NodeRecord) -> NodeInfo {
+        NodeInfo {
+            enode: enr.to_string(),
+            id: enr.id.to_string(),
+            ip: enr.address.to_string(),
+            listen_addr: enr.tcp_addr().to_string(),
+            name: "Reth".to_owned(),
+            ports: Ports { discovery: enr.tcp_port.into(), listener: enr.tcp_port.into() },
+            protocols: Protocols {
+                eth: Eth { difficulty: todo!(), genesis: todo!(), head: todo!(), network: todo!() },
+            },
+        }
+    }
+}
+
+/// Ports exposed by the node for discovery and listening.
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Ports {
+    /// Port exposed for node discovery.
+    pub discovery: u16,
+    /// Port exposed for listening.
+    pub listener: u16,
+}
+
+/// Networking protocols being run by the node.
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Protocols {
+    /// Information about the Ethereum Wire Protocol (ETH)
+    pub eth: Eth,
+}
+
+/// Information about the Ethereum Wire Protocol (ETH)
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Eth {
+    /// Total difficulty of the best chain.
+    pub difficulty: U256,
+    /// The hash of the genesis block.
+    pub genesis: String,
+    /// Hash of the latest block of the best chain.
+    pub head: String,
+    /// Network ID in base 10.
+    pub network: u64,
+}
```

Comment:
Do we have a way of pulling these from this crate already? Probably related to the syncing parts of Reth